### PR TITLE
[Bug] Fix html element hierarchy

### LINKED ISSUE #9999: 🐛 Element `p` is a child of a `span` element
## 🐛 Bug

Element `p` is a child of a `span` element.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/4858ebcd3b7ec8fcc5e5b0aaae92efaea52fa3c2/apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx#L393-L428

> [!NOTE]
> `<Button>` component has a child `span`.

## 🦋 Expected Behaviour

Element `p` should not be a child of element `span`.

## 📋 Steps to Reproduce

1. Navigate to `/admin/candidates/:candidate-id/application`'
2. Open dev tools elements tab
3. Inspect one of the buttons in the Screening and assessment table
4. Observe element `p` is a child of a `span` element

## 📸 Screenshot
Screen Shot 2024-04-05 at 10 41 23

## 🙋‍♀️ Proposed Solution

Change `p` to `span`.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] Element `p` not a child of element `span`

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #9999 

## 👋 Introduction

- Replace p with span tag

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Ensure p tag isn't used within span on Assessment table

### CODE REVIEW

**Comment by yonikid15** [comment]
> Close! There is one more `<p>` tag.

Nice catch! Fixed by [74d58a7](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10016/commits/74d58a7f54f135efda46a8ff5cf42f06e7b20c6c)

**Review by esizer — APPROVED** [review]
LGTM! :+1: 

Though, I'm wondering if we need a tag at all here. We should be able to just get away with a fragment `<>{text}</>`

File: apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx [hunk 1]
```diff
@@ -384,6 +384,42 @@ export const ScreeningDecisionDialog = ({
...
+                    initialValues.assessmentDecisionLevel,
+                  )
+                : commonMessages.notFound,
+            )}
+          </span>
+        ) : null}
+      </>
+    );
+  };
+
```
Comment (esizer) [inline apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:422]: Can we move this out to it's own component. We [shouldn't be defining components inside of other components](https://react.dev/learn/your-first-component).

**Review by esizer — CHANGES_REQUESTED** [review]

**Review by esizer — APPROVED** [review]
LGTM :+1:

**Comment by yonikid15** [comment]
@esizer Can you review this again 😅 
Matt and I made some changes again which should decrease the amount of code. Thanks @mnigh 👍

**Comment by mnigh** [comment]
> @esizer Can you review this again 😅 Matt and I made some changes again which should decrease the amount of code. Thanks @mnigh 👍

This PR was so good, @esizer approved it not once, not twice, but three times! 🥳

File: apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx [hunk 2]
```diff
@@ -397,48 +397,43 @@ export const ScreeningDecisionDialog = ({
...
-                      )}
-                    </p>
-                  ) : null}
-                </>
-              )}
-            </span>
-          ) : (
-            <p>
-              {poolSkill?.type === PoolSkillType.Nonessential &&
-              !experienceAttachedToSkill
```
Comment (esizer) [inline apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:437]: We seem to have lost this logic

**Review by esizer — CHANGES_REQUESTED** [review]
Looks like there is a regression :cry:

File: apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx (hunk 2 above)
Comment (yonikid15) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:437]: Fixed by [9c329d2](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10016/commits/9c329d25391d81d6f166a469710972dabb130f77)

Comment (esizer) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:437]: Hmm, not sure if that is quite right. We may need to get the gang together and work through this one :sweat:

Comment (yonikid15) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialog.tsx:437]: That was a test, you passed 🥇 This is the real fix [f4dca7a](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10016/commits/f4dca7a27ae8273fb650ded004b71acb7b9c9ff4)

**Review by esizer — APPROVED** [review]
LGTM! x4 :laughing:

**Pull request merged**